app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -0,0 +1,458 @@
+      tenant[:ems_ref] = t.id
+      tenant[:parent] = persister.cloud_tenants.lazy_find(t.try(:parent_id))
+    end
+  end
+
+  def miq_templates
+    collector.images.each do |i|
+      parent_server_uid = parse_image_parent_id(i)
+      image = persister.miq_templates.find_or_build(i.id)
+      image[:type] = "ManageIQ::Providers::Openstack::CloudManager::Template"
+      image[:uid_ems] = i.id
+      image[:ems_ref] = i.id
+      image[:name] = i.name
+      image[:vendor] = "openstack"
+      image[:raw_power_state] = "never"
+      image[:template] = true
+      image[:publicly_available] = public_image?(i)
+      image[:cloud_tenants] = image_tenants(i)
+      image[:location] = "unknown"
+      image[:hardware] = persister.hardwares.find_or_build(i.id)
+      image[:hardware][:vm_or_template] = persister.miq_templates.lazy_find(i.id)
+      image[:hardware][:bitness] = image_architecture(i)
+      image[:hardware][:disk_size_minimum] = (i.min_disk * 1.gigabyte)
+      image[:hardware][:memory_mb_minimum] = i.min_ram
+      image[:hardware][:root_device_type] = i.disk_format
+      image[:hardware][:size_on_disk] = i.size
+      image[:hardware][:virtualization_type] = i.properties.try(:[], 'hypervisor_type') || i.attributes['hypervisor_type']
+      image[:cloud_tenant] = persister.cloud_tenants.lazy_find(i.owner) if i.owner
+      image[:genealogy_parent] = persister.vms.lazy_find(parent_server_uid) unless parent_server_uid.nil?
+    end
+  end
+
+  def flavors
+    collector.flavors.each do |f|
+      make_flavor(f)
+    end
+  end
+
+  def make_flavor(f)
+    flavor = persister.flavors.find_or_build(f.id)
+    flavor[:type] = "ManageIQ::Providers::Openstack::CloudManager::Flavor"
+    flavor[:ems_ref] = f.id
+    flavor[:name] = f.name
+    flavor[:enabled] = !f.disabled
+    flavor[:cpus] = f.vcpus
+    flavor[:memory] = f.ram.megabytes
+    flavor[:publicly_available] = f.is_public
+    flavor[:root_disk_size] = f.disk.to_i.gigabytes
+    flavor[:swap_disk_size] = f.swap.to_i.megabytes
+    flavor[:ephemeral_disk_size] = f.ephemeral.nil? ? nil : f.ephemeral.to_i.gigabytes
+    flavor[:ephemeral_disk_count] = if f.ephemeral.nil?
+                                      nil
+                                    elsif f.ephemeral.to_i > 0
+                                      1
+                                    else
+                                      0
+                                    end
+    if f.is_public
+      # public flavors are associated with every tenant
+      flavor[:cloud_tenants] = collector.tenants.map { |tenant| persister.cloud_tenants.lazy_find(tenant.id) }
+    else
+      # Add tenants with access to the private flavor
+      flavor[:cloud_tenants] = collector.tenant_ids_with_flavor_access(f.id).map { |tenant_id| persister.cloud_tenants.lazy_find(tenant_id) }
+    end
+  end
+
+  def orchestration_stack_resources(stack)
+    resources = []
+    raw_resources = safe_list { stack.resources }
```

Comment:
would be nice to extract this call to Collector. Next step for the Collector will be to do the API collections in threads,  it should speed up these nested API call rapidly.